src/haar.jl

```diff
@@ -0,0 +1,159 @@
+        feats = 0
+        rects = 0
+    end
+
+    features = zeros(T, feats, rects)
+    for i = 1:feats
+        for j = 1:rects
+            features[i,j] = boxdiff(img, rectangle_features[i][j][1]:rectangle_features[i][j][3], rectangle_features[i][j][2]:rectangle_features[i][j][4])
+        end
+    end
+
+    output = (sum(features[:, 2:2:end], 2) - sum(features[:, 1:2:end], 2))[:]
+end
+
+"""
+```
+coordinates = haar_coordinates(height, width, feat)
+```
+
+Returns an array containing the coordinates of the Haar-like features of the specified type.
+
+Parameters:
+
+ -  height        = Height of the neighbourhood/window where the coordinates are computed
+ -  width         = Width of the neighbourhood/window where the coordinates are computed
+ -  feat          = A symbol specifying the type of the Haar-like feature to be found
+
+```
+    Currently, feat can take 5 values:
+
+    :x2 = Two rectangles along horizontal axis
+    :y2 = Two rectangles along vertical axis
+    :x3 = Three rectangles along horizontal axis
+    :y3 = Three rectangles along vertical axis
+    :xy4 = Four rectangles along horizontal and vertical axes
+```
+
+"""
+function haar_coordinates(h::Int, w::Int, feat::Symbol)
+    nrect = check_feature_type(feat)
+
+    a = Array{SVector{4,Int}}(nrect)
+    for i = 1:nrect
+        a[i] = SVector{4}(0,0,0,0)
+    end
+    rectangle_features = Array([a])
+
+    for I = 1:h
+        for J = 1:w
+            for i = 2:h
+                for j = 2:w
+                    if feat == :x2 && I + i - 2 <= h && J + 2*j - 3 <= w
+                        push!(rectangle_features, [SVector{4}(I, J, I + i - 2, J + j - 2), SVector{4}(I, J + j - 1, I + i - 2, J + 2*j - 3)])
+                    elseif feat == :y2 && I + 2*i - 3 <= h && J + j - 2 <= w
+                        push!(rectangle_features, [SVector{4}(I, J, I + i - 2, J + j - 2), SVector{4}(I + i - 1, J, I + 2*i - 3, J + j - 2)])
+                    elseif feat == :x3 && I + i - 2 <= h && J + 3*j - 4 <= w
+                        push!(rectangle_features, [SVector{4}(I, J, I + i - 2, J + j - 2), SVector{4}(I, J + j - 1, I + i - 2, J + 2*j - 3), SVector{4}(I, J + 2*j - 2, I + i - 2, J + 3*j - 4)])
+                    elseif feat == :y3 && I + 3*i - 4 <= h && J + j  - 2 <= w
+                        push!(rectangle_features, [SVector{4}(I, J, I + i - 2, J + j - 2), SVector{4}(I + i - 1, J, I + 2*i - 3, J + j - 2), SVector{4}(I + 2*i - 2, J, I + 3*i - 4, J + j - 2)])
+                    elseif feat == :xy4 && I + 2*i - 3 <= h && J + 2*j - 3 <= w
+                        push!(rectangle_features, [SVector{4}(I, J, I + i - 2, J + j - 2), SVector{4}(I, J + j - 1, I + i - 2, J + 2*j - 3), SVector{4}(I + i - 1, J + j - 1, I + 2*i - 3, J + 2*j - 3), SVector{4}(I + i - 1, J, I + 2*i - 3, J + j - 2)])
+                    end
+                end
+            end
+        end
+    end
+    deleteat!(rectangle_features, 1)
```

Comment:
I think we spoke about this on Slack too. You shouldn't have to initialize the array with a dummy entry only to have to delete it later.